base/inference.jl

```diff
@@ -52,20 +52,39 @@ const NF = NotFound()
 typealias LineNum Int
 typealias VarTable Array{Any,1}
 
+# The type of a variable load is either a value or an UndefVarError
 type VarState
     typ
     undef::Bool
     VarState(typ::ANY, undef::Bool) = new(typ, undef)
 end
 
-immutable Const
+# The type of a value might be constant
+type Const
```

Comment:
> stack allocating a pointer to a pointer

this can be very difficult, requires advanced escape analysis, and is often intractable due to all sorts of other aliasing concerns